### PR TITLE
New sample: Add 3d tiles layer

### PR DESCRIPTION
**Pull request by williambohrmann3**
# Description

Adds a new sample to the Layer category.

## Type of change

- New sample implementation

## Platforms tested on

- [x] WPF .NET 8
- [x] WPF Framework
- [x] WinUI
- [x] MAUI WinUI
- [x] MAUI Android
- [x] MAUI iOS
- [x] MAUI MacCatalyst

## Checklist

- [x] Self-review of changes
- [x] All changes work as expected on all affected platforms
- [x] There are no warnings related to changes
- [x] Code is commented and follows .NET conventions and standards
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] No unrelated changes have been made to any other code or project files
- [x] Screenshots are correct size and display in description tab (800 x 600 on Windows, MAUI mobile platforms use the MAUI Windows screenshot)

### CODE REVIEW

File: src/MAUI/Maui.Samples/Samples/Layers/Add3dTilesLayer/Add3dTilesLayer.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        tags: new[] { "3d tiles", "OGC", "OGC API", "layers", "scene", "service" })]
+    [ArcGIS.Samples.Shared.Attributes.OfflineData()]
+    public partial class Add3dTilesLayer
+    {
+        // Create a new elevation source from Terrain3D REST service.
+        private readonly ArcGISTiledElevationSource _elevationSource = new ArcGISTiledElevationSource
+            (new Uri("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"));
+
+        // Create a new scene with a dark gray basemap.
+        private readonly Scene _scene = new Scene(BasemapStyle.ArcGISDarkGray);
```
Comment (duffh) [inline src/MAUI/Maui.Samples/Samples/Layers/Add3dTilesLayer/Add3dTilesLayer.xaml.cs:28]: Does the `Scene` need to be a member variable?

Could we create this in `Initialize()`?

**Review by duffh — COMMENTED** [review]
Very minor nitpick overall it looks good.

File: src/MAUI/Maui.Samples/Samples/Layers/Add3dTilesLayer/Add3dTilesLayer.xaml.cs (hunk 1 above)
Comment (prathameshnarkhede) [reply src/MAUI/Maui.Samples/Samples/Layers/Add3dTilesLayer/Add3dTilesLayer.xaml.cs:28]: Agreed! We can move it to Initialize method as we are adding layers and elevation sources there. So we may want to make sure `scene` is initialized before performing operations on it.

**Review by prathameshnarkhede — APPROVED** [review]
LGTM!

**Pull request merged**